crates/pera-core/src/consensus_handler.rs

```diff
@@ -355,6 +356,63 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                             .stats
                             .inc_num_user_transactions(authority_index as usize);
                     }
+
+                    // If we receive a ProofMPCOutput transaction, verify that it's valid & create a system transaction
+                    // to store its output on the blockchain, so it will be available for the initiating user.
+                    if let ConsensusTransactionKind::SignatureMPCOutput(
+                        value,
+                        session_id,
+                        sender_address,
+                    ) = &transaction.kind
+                    {
+                        info!(
+                            "Received proof mpc output from authority {:?} for session {:?}",
+                            authority_index, session_id
+                        );
+
+                        let mut signature_mpc_manager = self.epoch_store.proof_mpc_manager.get();
+                        let is_valid_transaction = match signature_mpc_manager {
+                            Some(mpc_manager) => {
+                                let signature_mpc_manager = mpc_manager.lock().await;
+                                match signature_mpc_manager.try_verify_output(value, session_id) {
```

Comment:
I agree that this is a case that should be handled. Maybe Validators should check if MPC Messages are "last round
messages" and then compute the output based on those while they're synchronising?

Reply:
https://github.com/dwallet-labs/dwallet-network/pull/280/files#r1820448440

Reply:
what is this link? doesn't work for me. I don't get the answer, and don't feel comfotrable to resolve this. Please answer properly

Reply:
From what I understand, since all the validators process the MPC messages synchronously in the same order, the only scenario in which a validator will may not have an output at this point is if it's synchronizing. If this is the case, I created an issue so when validators synchronizing they should take no active actions, such as marking other validators as malicious --> https://github.com/dwallet-labs/dwallet-network/issues/311

Reply:
I see, I'm just not sure that not taking an active action is sufficient - the output should be approved by 2/3 of stake, so a synchronizing validator will not be able to approve. But in that case you say that there won't be enough approvals, and the output won't be written to the state. Ok then #311 is enough